### PR TITLE
Improve Algolia search ranking & results

### PR DESCRIPTION
**Pull request by kmaid**
This PR is to implement the website side of https://www.algolia.com/doc/guides/sending-events/connectors/google-tag-manager/

Algolia Client and GTM are both tagged with UserId so queries can be related to a user.
Result lists rendered from algolia data should be tagged with the index they were run on

- Events
- Users
- Research Outputs

Hits should be tagged with QueryId, Position and Object ID which is the same as our ID's 
I have written very high level unit tests to ensure tags are present on the relevant result lists to prevent regressions.

I have also refactored the algolia scripts to be within the algolia package.

### CODE REVIEW

File: apps/asap-cli/package.json [hunk 1]
```diff
@@ -7,6 +7,12 @@
     "node": "18.x"
   },
   "scripts": {
+    "algolia:delete-index": "ts-node --transpile-only src/cli.ts algolia:delete-index",
```
Comment (gabiayako) [inline apps/asap-cli/package.json:10]: isn't `TS_NODE_COMPILER_OPTIONS='{\"module\": \"commonjs\"}` necessary in some scripts like it was in the root package.json?

File: packages/react-components/src/templates/SharedResearchList.tsx [hunk 2]
```diff
@@ -18,18 +19,26 @@ type SharedResearchListProps = Omit<
...
+} & Pick<ComponentProps<typeof AlgoliaHit>, 'algoliaQueryId'>;
 
 const SharedResearchList: React.FC<SharedResearchListProps> = ({
   researchOutputs,
+  algoliaQueryId,
   ...cardListProps
 }) => (
   <ResultList {...cardListProps}>
     {cardListProps.isListView ? (
       <SharedResearchListCard researchOutputs={researchOutputs} />
```
Comment (gabiayako) [inline packages/react-components/src/templates/SharedResearchList.tsx:31]: isn't `AlgoliaHit` necessary here as well?

**Review by gabiayako — APPROVED** [review]

File: apps/asap-cli/package.json (hunk 1 above)
Comment (kmaid) [reply apps/asap-cli/package.json:10]: No, we moved it into tsconfig in the TS_Node section for the entire app

File: packages/react-components/src/templates/SharedResearchList.tsx (hunk 2 above)
Comment (kmaid) [reply packages/react-components/src/templates/SharedResearchList.tsx:31]: Good catch.

**Pull request merged**